mod/metrics/set/abstract/01_metric_child.rb

```diff
@@ -14,15 +14,34 @@ def self.included host_class
 end
 
 def metric_part full_name=name
-  full_name.to_name.parts[0..(-1 - generation)].join("+")
+  parts = full_name.to_name.parts
+  metric_parts = parts[0..(-1 - generation)]
+  if metric_parts.size > 2 && company_with_plus?(parts)
+    metric_parts[0..1].join Cardname.joint
+  else
+    metric_parts.join Cardname.joint
+  end
```

Comment:
maybe make `metric_parts` a method, and this just becomes  `metric_parts.join Cardname.joint`